com.unity.render-pipelines.high-definition/Runtime/Material/Decal/DecalUtilities.hlsl

```diff
@@ -253,25 +247,9 @@ DecalSurfaceData GetDecalSurfaceData(PositionInputs posInput, inout float alpha)
 #endif // LIGHTLOOP_DISABLE_TILE_AND_CLUSTER
 
         uint s_decalIdx = v_decalIdx;
-
-#if SCALARIZE_LIGHT_LOOP
-
-        if (!fastPath)
-        {
-            // If we are not in fast path, v_lightIdx is not scalar, so we need to query the Min value across the wave.
-            s_decalIdx = WaveActiveMin(v_decalIdx);
-            // If WaveActiveMin returns 0xffffffff it means that all lanes are actually dead, so we can safely ignore the loop and move forward.
-            // This could happen as an helper lane could reach this point, hence having a valid v_lightIdx, but their values will be ignored by the WaveActiveMin
-            if (s_decalIdx == -1)
-            {
-                break;
-            }
-        }
-        // Note that the WaveReadLaneFirst should not be needed, but the compiler might insist in putting the result in VGPR.
-        // However, we are certain at this point that the index is scalar.
-        s_decalIdx = WaveReadLaneFirst(s_decalIdx);
-
-#endif // SCALARIZE_LIGHT_LOOP
+        uint s_decalIdx = ScalarizeElementIndex(v_decalIdx, fastPath);
+        if (s_decalIdx == -1)
```

Comment:
guess we should keep this comment?

            // If WaveActiveMin returns 0xffffffff it means that all lanes are actually dead, so we can safely ignore the loop and move forward.	
            // This could happen as an helper lane could reach this point, hence having a valid v_lightIdx, but their values will be ignored by the WaveActiveMin

?

Reply:
nevermind